sdk/core/core-tracing/src/createSpan.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Span, SpanOptions, SpanKind } from "@opentelemetry/api";
+import { getTracer } from "../src/tracerProxy";
+
+/**
+ * Tracing options to set on an operation.
+ * @hidden
+ */
+export interface OperationTracingOptionsLike {
+  /**
+   * OpenTelemetry SpanOptions used to create a span when tracing is enabled.
+   */
+  spanOptions?: SpanOptions;
+}
+
+/**
+ * Configuration for creating a new Tracing Span
+ * @hidden
+ */
+export interface SpanConfig {
+  /**
+   * Package name prefix
+   */
+  packagePrefix: string;
+  /**
+   * Service namespace
+   */
+  namespace: string;
+}
+
+/**
+ * Creates a function called createSpan to create spans using the global tracer.
+ * @hidden
+ * @param spanConfig - The name of the operation being performed.
+ * @param tracingOptions - The options for the underlying http request.
+ */
+export function createSpanFunction({ packagePrefix, namespace }: SpanConfig) {
+  return function<T extends { tracingOptions?: OperationTracingOptionsLike } | undefined>(
+    operationName: string,
+    operationOptions: T
+  ): { span: Span; updatedOptions: T } {
+    const tracer = getTracer();
+    const tracingOptions = operationOptions?.tracingOptions || {};
+    const spanOptions: SpanOptions = {
+      ...tracingOptions.spanOptions,
+      kind: SpanKind.INTERNAL
+    };
+
+    const span = tracer.startSpan(`${packagePrefix}.${operationName}`, spanOptions);
+
+    span.setAttribute("az.namespace", namespace);
+
+    let newSpanOptions = tracingOptions.spanOptions || {};
+    if (span.isRecording()) {
+      newSpanOptions = {
+        ...tracingOptions.spanOptions,
+        parent: span.context(),
+        attributes: {
+          ...spanOptions.attributes,
+          "az.namespace": namespace
+        }
+      };
+    }
+
+    const newTracingOptions: OperationTracingOptionsLike = {
+      ...tracingOptions,
+      spanOptions: newSpanOptions
+      // TODO: .context soon.
```

Comment:
do we even need to worry about .context stuff? Won't it default to the current context?

Reply:
It would default to context.active() if it's undefined but I assume if they pass us a context we should use it and ignore context.active(), right?

Reply:
I'm fine with us passing it along, but we shouldn't default it to anything ourselves

Reply:
Yeah, that's the plan for sure. :)